test/scripts/dev.test.js

```diff
@@ -920,3 +920,13 @@ describe('port unavailable', () => {
     expect(resultUrl).toBe('https://localhost:99')
   })
 })
+
+test('No backend is present', async () => {
+  global.loadFs(vol, 'sample-app')
+  mockAIOConfig.get.mockReturnValue(global.fakeConfig.tvm)
+  vol.unlinkSync('./manifest.yml')
+
+  const scripts = await AppScripts()
+  await scripts.runDev()
+  expect(vol.existsSync('/web-src/src/config.json')).toEqual(true)
```

Comment:
Here I would put this test as part of a describe (ui only) and add a few more tests:
- ui server has started
- no ow server started
- no build & deploy called
- only ui config in vscode file
- config.json = {}
- scripts._config.hasBackend = false